$NetBSD: patch-ae,v 1.4 2009/02/22 15:40:08 tnn Exp $

Thousands grouping doesn't work correctly on NetBSD when used with %f.
See PR lib/40714.

--- gtk/util.c.orig	2009-02-13 17:40:54.000000000 +0100
+++ gtk/util.c
@@ -58,11 +58,11 @@ tr_strlratio( char * buf,
     else if( (int)ratio == TR_RATIO_INF )
         g_strlcpy( buf, "\xE2\x88\x9E", buflen );
     else if( ratio < 10.0 )
-        g_snprintf( buf, buflen, "%'.2f", ratio );
+        g_snprintf( buf, buflen, "%.2f", ratio );
     else if( ratio < 100.0 )
-        g_snprintf( buf, buflen, "%'.1f", ratio );
+        g_snprintf( buf, buflen, "%.1f", ratio );
     else
-        g_snprintf( buf, buflen, "%'.0f", ratio );
+        g_snprintf( buf, buflen, "%.0f", ratio );
     return buf;
 }
 
@@ -95,17 +95,17 @@ tr_strlsize( char *  buf,
         if( size < (guint64)MEGABYTE_FACTOR )
         {
             displayed_size = (gdouble) size / KILOBYTE_FACTOR;
-            g_snprintf( buf, buflen, _( "%'.1f KB" ), displayed_size );
+            g_snprintf( buf, buflen, _( "%.1f KB" ), displayed_size );
         }
         else if( size < (guint64)GIGABYTE_FACTOR )
         {
             displayed_size = (gdouble) size / MEGABYTE_FACTOR;
-            g_snprintf( buf, buflen, _( "%'.1f MB" ), displayed_size );
+            g_snprintf( buf, buflen, _( "%.1f MB" ), displayed_size );
         }
         else
         {
             displayed_size = (gdouble) size / GIGABYTE_FACTOR;
-            g_snprintf( buf, buflen, _( "%'.1f GB" ), displayed_size );
+            g_snprintf( buf, buflen, _( "%.1f GB" ), displayed_size );
         }
     }
 #endif
@@ -120,13 +120,13 @@ tr_strlspeed( char * buf,
     const double speed = kb_sec;
 
     if( speed < 1000.0 )  /* 0.0 KB to 999.9 KB */
-        g_snprintf( buf, buflen, _( "%'.1f KB/s" ), speed );
+        g_snprintf( buf, buflen, _( "%.1f KB/s" ), speed );
     else if( speed < 102400.0 ) /* 0.98 MB to 99.99 MB */
-        g_snprintf( buf, buflen, _( "%'.2f MB/s" ), ( speed / 1024 ) );
+        g_snprintf( buf, buflen, _( "%.2f MB/s" ), ( speed / 1024 ) );
     else if( speed < 1024000.0 ) /* 100.0 MB to 999.9 MB */
-        g_snprintf( buf, buflen, _( "%'.1f MB/s" ), ( speed / 1024 ) );
+        g_snprintf( buf, buflen, _( "%.1f MB/s" ), ( speed / 1024 ) );
     else /* insane speeds */
-        g_snprintf( buf, buflen, _( "%'.2f GB/s" ), ( speed / 1048576 ) );
+        g_snprintf( buf, buflen, _( "%.2f GB/s" ), ( speed / 1048576 ) );
 
     return buf;
 }
